var/ramble/repos/builtin/modifiers/gcp-metadata/modifier.py

```diff
@@ -24,23 +24,62 @@ class GcpMetadata(BasicModifier):
     mode('standard', description='Standard execution mode')
     default_mode('standard')
 
-    variable_modification('log', '{experiment_run_dir}/gcp-metadata.log', method='set', modes=['standard'])
-    archive_pattern('{log}')
+    executable_modifier('gcp_metadata_exec')
 
-    register_builtin('gcp_metadata_exec')
+    def gcp_metadata_exec(self, executable_name, executable, app_inst=None):
+        from ramble.util.executable import CommandExecutable
+        post_cmds = []
+        pre_cmds = []
 
-    def gcp_metadata_exec(self):
-        machine_type = 'curl -s -w "\\n" "http://metadata.google.internal/computeMetadata/v1/instance/machine-type" -H "Metadata-Flavor: Google" >> {log}'
+        payloads = [
+            # end point, use_mpi
+            ('machine-type', False),
+            ('image', False),
+            ('hostname', False),
+            ('id', True),  # True since we want the gid of every node
+        ]
 
-        image = 'curl -s -w "\\n" "http://metadata.google.internal/computeMetadata/v1/instance/image" -H "Metadata-Flavor: Google" >> {log}'
+        rm_template = []
+        for end_point, _ in payloads:
+            rm_template.append(f'rm "{{experiment_run_dir}}/gcp-metadata.{end_point}.log"')
 
-        gid = 'curl -s -w "=GID\\n" "http://metadata.google.internal/computeMetadata/v1/instance/id" -H "Metadata-Flavor: Google" >> {log}'
+        pre_cmds.append(
```

Comment:
Instead of explicitly removing the templates, could you just change the `output_capture` operator to `>`.

Either is fine, I was more just curious if there's a reason to explicitly `rm` it every time.

Reply:
I agree, good change. Initially I was thinking of supporting other ways to request the data in parallel and was scared that would cause a data race.. but honestly I don't think that's possible. Will update